src/com/owncloud/android/files/FileOperationsHelper.java

```diff
@@ -88,13 +93,28 @@ public void openFile(OCFile file) {
             }
             
             Intent chooserIntent;
+            List<ResolveInfo> launchables;
             if (intentForGuessedMimeType != null) {
                 chooserIntent = Intent.createChooser(intentForGuessedMimeType, mFileActivity.getString(R.string.actionbar_open_with));
+                launchables = mFileActivity.getPackageManager().queryIntentActivities(intentForGuessedMimeType, PackageManager.GET_INTENT_FILTERS);
             } else {
                 chooserIntent = Intent.createChooser(intentForSavedMimeType, mFileActivity.getString(R.string.actionbar_open_with));
+                launchables = mFileActivity.getPackageManager().queryIntentActivities(intentForSavedMimeType, PackageManager.GET_INTENT_FILTERS);
             }
 
-            mFileActivity.startActivity(chooserIntent);
+            if(launchables != null && launchables.size() > 0) {
+                try {
+                    mFileActivity.startActivity(chooserIntent);
+                } catch (ActivityNotFoundException anfe) {
+                    Toast.makeText(mFileActivity.getApplicationContext(),
+                            R.string.file_list_no_app_for_file_type, Toast.LENGTH_SHORT)
+                            .show();
+                }
+            } else {
+                Toast.makeText(mFileActivity.getApplicationContext(),
+                        R.string.file_list_no_app_for_file_type, Toast.LENGTH_SHORT)
+                        .show();
+            }
```

Comment:
Same here.
Maybe a small function "showNoAppToast()" and avoid the code duplication.